pkg/eks/tasks.go

```diff
@@ -87,3 +119,23 @@ func (c *ClusterProvider) appendCreateTasksForIAMServiceAccounts(cfg *api.Cluste
 	newTasks.IsSubTask = true
 	tasks.Append(newTasks)
 }
+
+func (c *ClusterProvider) maybeAppendTasksForEndpointAccessUpdates(cfg *api.ClusterConfig, newTasks, tasks *manager.TaskTree) {
+	// if a cluster config doesn't have the default api endpoint access, append a new task
+	// so that we update the cluster with the new access configuration.  This is a
+	// non-CloudFormation context, so we create a task to send it through the EKS API.
+	// A caveat is that sending the default endpoint parameters for a cluster as an update will
+	// return an error from the EKS API, so we must check for this before sending the request.
+	if cfg.HasClusterEndpointAccess() && endpointsEqual(*cfg.VPC.ClusterEndpoints, *api.ClusterEndpointAccessDefaults()) {
+		logger.Info(DefaultEndpointsMsg(cfg))
+		logger.Info(UpdateEndpointsMsg(cfg))
+	} else {
+		logger.Info(CustomEndpointsMsg(cfg))
+		newTasks.Append(&clusterConfigTask{
+			info: "update cluster VPC endpoint access configuration",
+			spec: cfg,
+			call: c.UpdateClusterConfigForEndpoints,
+		})
+		tasks.Append(newTasks)
```

Comment:
i think this should be moved back to where it was

Reply:
moved.